app/controllers/hosts_controller.rb

```diff
@@ -261,6 +267,18 @@ def power
     process_error :redirect => :back, :error_msg => _("Failed to %{action} %{host}: %{e}") % { :action => _(params[:power_action]), :host => @host, :e => e }
   end
 
+  def get_power_state
+    result = {:id => @host.id, :state => HOST_POWER[:na]}
+    if @host.supports_power?
+      result[:state] = HOST_POWER[@host.supports_power_and_running? ? :on : :off]
+    end
+    render :json => result
+  rescue => e
+    Foreman::Logging.exception("Failed to fetch power status", e)
+    result[:state]= HOST_POWER[:na]
+    render :json => result
```

Comment:
I think it makes sense to return a status different then 200 when an error occurs